### PR TITLE
Feature/more changes

### PR DESCRIPTION
**Pull request by beseven**

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
Odd capitalisation going on with text in braces (e.g. https://staging.c2s.nhschoices.net/book-a-gp-appointment/results?search=Albany+House+Medical+Centre)
Search including '(' throws error.
Name on Profile page is not processed (e.g. https://staging.c2s.nhschoices.net/gp-surgeries/42499/book-appointment)

**Comment by beseven** [comment]
Based on a conversation with @markysoft, I'll tackle the problems here first and then we can add them to the etl later.

**Comment by beseven** [comment]
@neilbmclaughlin - will tackle the "(" issue when refactoring the mongo function if that is ok?

**Comment by beseven** [comment]
Rebased and re-pushed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

File: app/lib/utils/properCapitalize.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      if (str.charAt(0).match(/[^a-zA-Z\d]/)) {
+        return capitalizeWithSpecialCharacter(str);
+      }
+    return capitalize(str);
+  });
+
+  return stringArray.join(' ');
+}
+
+module.exports = titleize;
```
Comment (markysoft) [inline app/lib/utils/properCapitalize.js]: Doesn't correctly format names starting **O'** such as O'Toole.
http://gp-finder-pr-21.dev.c2s.nhschoices.net/book-a-gp-appointment/results?search=tool

It's also changing the case on abbreviations such as DMC, but don't think that can be detected via code.

File: test/unit/lib/utils/properCapitalize.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    expect(result).to.be.equal(expectedResult);
+  });
+  it('for a lowercase string that starts with a non-alphanumeric character should capitalize only the first letter', () => {
+    const str = 'clapham (park group) practice';
+    const expectedResult = 'Clapham (Park Group) Practice';
+    const result = properCapitalize(str);
+
+    expect(result).to.be.equal(expectedResult);
+  });
+  it('for an uppercase string should leave the first letter alone and downcase the rest', () => {
```
Comment (markysoft) [inline test/unit/lib/utils/properCapitalize.js]: The other tests have spaces between the 'describe' blocks

File: app/lib/utils/properCapitalize.js (hunk 1 above)
Comment (beseven) [inline app/lib/utils/properCapitalize.js]: Will look at this. I need to make sure it won't affect things like O'Toole's.

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-21.dev.c2s.nhschoices.net](http://gp-finder-pr-21.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
Dismissing Marks review as suggested changes regarding apostrophes have been made and issue of acronyms raised as issue #30.

**Pull request merged**